spec/fakefs_helper.rb

```diff
@@ -0,0 +1,50 @@
+#==============================================================================
+# Copyright (C) 2017 Stephen F. Norledge and Alces Software Ltd.
+#
+# This file/package is part of Alces Metalware.
+#
+# Alces Metalware is free software: you can redistribute it and/or
+# modify it under the terms of the GNU Affero General Public License
+# as published by the Free Software Foundation, either version 3 of
+# the License, or (at your option) any later version.
+#
+# Alces Metalware is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this package.  If not, see <http://www.gnu.org/licenses/>.
+#
+# For more information on the Alces Metalware, please visit:
+# https://github.com/alces-software/metalware
+#==============================================================================
+require 'fakefs/safe'
+
+class FakeFSHelper
+  def initialize(metal_config)
+    @metal_config = metal_config
+    FakeFS.deactivate!
+    FakeFS.clear!
+  end
+
+  def run(&block)
+    FakeFS do yield end
+  end
```

Comment:
Requiring `FakeFS` is error prone. For example `require 'fakefs` will immediately start faking the FS. That is why `require 'fakefs/safe'` needs to be used instead. 

To prevent this ever becoming an issue, I wanted to abstract away all of the `FakeFS` commands we will use regularly into `FakeFSHelper`. This way we don't need to worry about `requring 'fakefs'` into the test themselves. 

tl;dr; We don't absolutely need it, but it might make our lives easier if we have it anyway